plugins/woocommerce/src/Utilities/PluginUtil.php

```diff
@@ -195,20 +199,29 @@ public function generate_incompatible_plugin_feature_warning( string $feature_id
 		$incompatibles      = array_filter( $incompatibles, 'is_plugin_active' );
 		$incompatibles      = array_values( array_diff( $incompatibles, $this->get_plugins_excluded_from_compatibility_ui() ) );
 		$incompatible_count = count( $incompatibles );
+
+		$feature_warnings = array();
+		if ( 'custom_order_tables' === $feature_id && 'yes' === get_option( 'woocommerce_api_enabled' ) ) {
+			if ( is_plugin_active( 'woocommerce-legacy-rest-api/woocommerce-legacy-rest-api.php' ) ) {
+				$feature_warnings[] = __( "⚠ <b>The Legacy REST API plugin is installed and active on this site.</b> Please be aware that the WooCommerce Legacy REST API is <b>not</b> compatible with HPOS.\n", 'woocommerce' );
```

Comment:
Should we link to the plugin page/legacy API setting page here, I don't suppose merchants seeing this will immediately know what legacy API is?